src/lib/comms/include/sol-http.h

```diff
@@ -132,7 +135,18 @@ struct sol_http_response {
     int response_code;
 };
 
-#define SOL_HTTP_RESPONSE_CHECK_API(response_, ...) \
+struct sol_http_url {
+    struct sol_str_slice scheme;
+    struct sol_str_slice user;
+    struct sol_str_slice password;
+    struct sol_str_slice host;
+    struct sol_str_slice path;
+    struct sol_str_slice query;
+    struct sol_str_slice fragment;
+    int port; //If < 0 ignore
```

Comment:
int32_t... and I guess 0 is already not a valid port, so you can make it `uint32_t` and say `if == 0, ignore`